bot/config.py

```diff
@@ -1,26 +1,30 @@
 import yaml
-import dotenv
+import os
 from pathlib import Path
+import ast
+
+from dotenv import load_dotenv
+
+
 
 config_dir = Path(__file__).parent.parent.resolve() / "config"
+load_dotenv(dotenv_path=config_dir / "config.env")
 
 # load yaml config
 with open(config_dir / "config.yml", 'r') as f:
     config_yaml = yaml.safe_load(f)
 
-# load .env config
-config_env = dotenv.dotenv_values(config_dir / "config.env")
-
 # config parameters
-telegram_token = config_yaml["telegram_token"]
-openai_api_key = config_yaml["openai_api_key"]
+openai_api_key = os.environ["OPENAI_API_KEY"]
+telegram_token = os.environ["TELEGRAM_TOKEN"]
 use_chatgpt_api = config_yaml.get("use_chatgpt_api", True)
-allowed_telegram_usernames = config_yaml["allowed_telegram_usernames"]
+env_allowed_users = os.environ.get('ALLOWED_TELEGRAM_USERNAMES', '')
+allowed_telegram_usernames = (lambda s: s.split(',') if s else [])(os.environ.get('ALLOWED_TELEGRAM_USERNAMES', ''))
 new_dialog_timeout = config_yaml["new_dialog_timeout"]
```

Comment:
this does not have a default value. it should be 

`new_dialog_timeout = config_yaml.get("new_dialog_timeout", 600)`